### DISCUSSION TITLE
Which methods are equivalent to auth.api.setAuthCookie(req, res) and auth.api.getUserByCookie(req) in Supabase JS library v2?

### DISCUSSION THREAD

**Discussion by taishikato** [supabase/supabase#10051] [Questions] [answered]
Hi, Supabase team and devs!

I encountered some confusion when I was trying to set cookie stuff on Next.js
I couldn't find the methods below. I believe this is because I am using @supabase/supabase-js v2 library.

* `supabase.auth.api.setAuthCookie(req, res)`
* `supabase.auth.api.getUserByCookie(req)`

I read the blog: https://supabase.com/blog/supabase-js-v2#auth-admin-methods
It says `supabase.auth.api` has been moved to `supabase.auth.admin`, however, in `supabase.auth.admin`, I couldn't find the method I need.

<img width="660" alt="CleanShot 2022-11-02 at 23 16 51@2x" src="https://user-images.githubusercontent.com/980588/199658056-9ef18445-2222-491c-94ba-79d3971b8454.png">

Any help?

Thank you!

**Comment by thorwebdev** [comment, marked as answer]
These methods have been removed. For nextjs you can use the Auth Helpers to help you manage cookies: https://supabase.com/docs/guides/auth/auth-helpers/nextjs

If you prefer not to use the auth helpers package you can follow this approach: https://supabase.com/docs/guides/auth/server-side-rendering

**Reply by taishikato** [reply]
@thorwebdev Thanks!
BTW, the sample code is messed up.
<img width="794" alt="CleanShot 2022-11-03 at 11 58 19@2x" src="https://user-images.githubusercontent.com/980588/199810723-967fd77a-bcdb-47b8-a507-363a7ee802a4.png">

**Reply by thorwebdev** [reply]
Thanks, this has since been fixed 👍
